commons/src/test/java/org/eclipse/kapua/commons/service/event/store/internal/EventStoreRecordImplTest.java

```diff
@@ -0,0 +1,464 @@
+        assertEquals("Expected and actual values should be the same.", new KapuaEid(BigInteger.TEN), eventStoreRecordImpl.getUserId());
+        assertEquals("Expected and actual values should be the same.", "service", eventStoreRecordImpl.getService());
+        assertEquals("Expected and actual values should be the same.", "entityType", eventStoreRecordImpl.getEntityType());
+        assertEquals("Expected and actual values should be the same.", new KapuaEid(BigInteger.TEN), eventStoreRecordImpl.getEntityId());
+        assertEquals("Expected and actual values should be the same.", "operation", eventStoreRecordImpl.getOperation());
+        assertEquals("Expected and actual values should be the same.", "input", eventStoreRecordImpl.getInputs());
+        assertEquals("Expected and actual values should be the same.", "output", eventStoreRecordImpl.getOutputs());
+        assertEquals("Expected and actual values should be the same.", ServiceEvent.EventStatus.TRIGGERED, eventStoreRecordImpl.getStatus());
+        assertEquals("Expected and actual values should be the same.", "note", eventStoreRecordImpl.getNote());
+    }
+
+    @Test
+    public void setAndGetContextIdTest() throws KapuaException {
+        String contextId = "ContextId";
+        KapuaId[] scopeIdList = {null, new KapuaIdStatic(BigInteger.ONE), new KapuaIdStatic(BigInteger.TEN), new KapuaIdStatic(BigInteger.ZERO)};
+
+        for (KapuaId scopeId : scopeIdList) {
+            EventStoreRecordImpl eventStoreRecordImpl1 = new EventStoreRecordImpl(scopeId);
+            assertNull("Null expected.", eventStoreRecordImpl1.getContextId());
+            eventStoreRecordImpl1.setContextId(contextId);
+            assertEquals("Expected and actual values should be the same.", contextId, eventStoreRecordImpl1.getContextId());
+            eventStoreRecordImpl1.setContextId(null);
+            assertNull("Null expected.", eventStoreRecordImpl1.getContextId());
+        }
+
+        EventStoreRecordImpl eventStoreRecordImpl2 = new EventStoreRecordImpl();
+        assertNull("Null expected.", eventStoreRecordImpl2.getContextId());
+        eventStoreRecordImpl2.setContextId(contextId);
+        assertEquals("Expected and actual values should be the same.", contextId, eventStoreRecordImpl2.getContextId());
+        eventStoreRecordImpl2.setContextId(null);
+        assertNull("Null expected.", eventStoreRecordImpl2.getContextId());
+
+        EventStoreRecordImpl eventStoreRecordImpl3 = new EventStoreRecordImpl(eventStoreRecord);
+        assertEquals("Expected and actual values should be the same.", "contextId", eventStoreRecordImpl3.getContextId());
+        eventStoreRecordImpl3.setContextId(contextId);
+        assertEquals("Expected and actual values should be the same.", contextId, eventStoreRecordImpl3.getContextId());
+        eventStoreRecordImpl3.setContextId(null);
+        assertNull("Null expected.", eventStoreRecordImpl3.getContextId());
```

Comment:
The clone method has been already been tested in `eventStoreRecordImplTest3`, if I'm correct.
So also this is not needed.

Reply:
@Coduz here we are checking if the contextId is null and in the "eventStoreRecordImplTest3" we check if the expected and actual values are the same, so these are two different cases. If I am seeing right.